### PR TITLE
Update block design principles with a new section on how to group controls

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

This takes a first stab at creating new documentation for block toolbar group best practices.

See also #25983.

## Screenshots 

Screenshot 2021-03-12 at 11 29 17

There's also a new "Do and don't" section for the block toolbar:

I also updated the placeholder screenshot while I was there:

Screenshot 2021-03-12 at 11 29 32

The rest of the screenshots also need updating, but one step at a time.

### CODE REVIEW

**Review by aristath — APPROVED** [review]

**Comment by jasmussen** [comment]
💯 So fast! Thank you Ari!

**Comment by mtias** [comment]
Awesome, this is a great example of what we need to codify more, thanks!

File: docs/how-to-guides/designers/block-design.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ Since the block itself represents what will actually appear on the site, interac
 
 Basic block settings won’t always make sense in the context of the placeholder/content UI. As a secondary option, options that are critical to the functionality of a block can live in the block toolbar. The Block Toolbar is still highly contextual and visible on all screen sizes. One notable constraint with the Block Toolbar is that it is icon-based UI, so any controls that live in the Block Toolbar need to be ones that can effectively be communicated via an icon or icon group.
 
+### Group Block Toolbar controls with related items
+
+The Block Toolbar groups controls in segments, hierarchically. The first segment contains block controls, such as the block switcher, the drag handle, and the mover control. The second group commonly contains tools, followed by inline formatting, and the "More" menu. Optionally "Meta" or "Other" groups can separate some tools in their own segment.
```
Comment (mtias) [inline docs/how-to-guides/designers/block-design.md:20]: Some suggestions:

> The first segment contains block type controls, such as the block switcher, the drag handle, and the mover control. The second group contains common and specific block tools that affect the entire block, followed by inline...

Comment (jasmussen) [reply docs/how-to-guides/designers/block-design.md]: Perfect 👌

**Review by mtias — APPROVED** [review]

**Comment by noisysocks** [comment]
Could we link each of the hierarchy names (Block, Tools, Meta, etc.) to the exact values that one should pass to `<BlockControls group="">`? Or document that someplace else? I'm struggling to figure out how to translate this excellent guideline into code.

**Pull request merged**